### PR TITLE
Scope SQLA session so object lifespans rendering for task history

### PR DESCRIPTION
**Pull request by bsdz**
## Description

For SQLAlchemy >1.4 it seems setting `expire_on_commit` to `False` is less forgiving that older versions. This causes the following exception to be thrown when visiting the "/history" tab.

```python
sqlalchemy.orm.exc.DetachedInstanceError: Parent instance <TaskRecord at 0x7f943e167190> is not bound to a Session; lazy load operation of attribute 'parameters' cannot proceed (Background on this error at: http://sqlalche.me/e/14/bhk3)
```

## Motivation and Context
This resolves issue https://github.com/spotify/luigi/issues/3006.

## Have you tested this? If so, how?
I ran my jobs with this code and it works for me when using SQLAlchemy 1.4.1.

### CODE REVIEW

**Review by dlstadther — APPROVED** [review]

**Comment by robsalasco** [comment]
would be nice if this patch can be merged in the next version 😞

**Comment by tashrifbillah** [comment]
I am giving this PR a go. The current release `3.0.3` did not show `/history` for me in my CentOS 7 Linux machine. After manually affecting these changes, now I can see `/history`. Thanks for the work @bsdz .

cc @dstandish

**Comment by pmhaddad** [comment]
Commenting just to make it easier to keep an eye on this **PR** and related Issue, since we are also being affected by this at the moment :+1:

**Comment by robsalasco** [comment]
ping @Tarrasch

**Comment by tashrifbillah** [comment]
> ping @Tarrasch

He is not the maintainer anymore ...

**Comment by dlstadther** [comment]
I'm good to merge this if we can get the feature branch updated with master and tests passing

**Comment by tashrifbillah** [comment]
Noting the current working combination until this is merged--`pip install luigi==2.8.12 sqlalchemy==1.3.16`

**Comment by tashrifbillah** [comment]
> I'm good to merge this if we can get the feature branch updated with master and tests passing

ping @bsdz

**Comment by bsdz** [comment]
> > I'm good to merge this if we can get the feature branch updated with master and tests passing
> 
> ping @bsdz 

Very sorry I'm no longer using this project and haven't time to update this fork.

**Comment by tashrifbillah** [comment]
Maybe another of us can volunteer to update his fork ...

**Comment by dlstadther** [comment]
Replaced by #3153

**Pull request closed without merging**
